public/app/listeners/createExecution.js

```diff
@@ -0,0 +1,37 @@
+const ObjectId = require('bson-objectid');
+const { EXECUTIONS_COLLECTION } = require('../db');
+const { CREATE_EXECUTION_CHANNEL } = require('../../shared/channels');
+const { EXECUTION_STATUSES } = require('../config/config');
+const { ERROR_GENERAL } = require('../../shared/errors');
+const {
+  CREATE_EXECUTION_SUCCESS,
+  CREATE_EXECUTION_ERROR,
+} = require('../../shared/types');
+
+const createExecution = (mainWindow, db) => async (
+  event,
+  { algorithmId, datasetId },
+) => {
+  try {
+    // add execution
+    const execution = {
+      id: ObjectId().str,
+      algorithmId,
+      datasetId,
+      executedAt: Date.now(),
+      status: EXECUTION_STATUSES.PENDING,
```

Comment:
Indeed nothing is displayed for pending executions. The whole goal of pending is to let the possibility (in the future) to start the execution at a later time point instead of running the execution right away. Maybe it will ease the process for pause/start a process (but I am not yet sure how it would work). So I was thinking about displaying a start button for pending executions, let me know what you think @hagoptaminian and @Painguin as well.

Reply:
I think it is good to automatically start the execution as it is added, but we could have a `start now` checkbox if we want to allow starting the execution later. And then have a `start`, `restart`, `pause` or `unpause` button depending on the state of the execution.

Reply:
interesting idea. let's think about it. after clicking Execute, why would a user need to 're-confirm' the execution (hypothetically marked as pending)? but I am open to it. in any case since an execution now moves from pending to running (after the changes to `executeAlgorithm.js`), it's less of an issue...

Reply:
I'll make an issue with it. We talk about it tomorrow.